actors/builtin/multisig/multisig_test.go

```diff
@@ -165,6 +167,154 @@ func TestPropose(t *testing.T) {
 		})
 
 		actor.assertTransactions(rt)
+		rt.Verify()
+	})
+}
+
+func TestApprove(t *testing.T) {
+	actor := msActorHarness{multisig.MultiSigActor{}, t}
+
+	receiver := newIDAddr(t, 100)
+	anne := newIDAddr(t, 101)
+	bob := newIDAddr(t, 102)
+	chuck := newIDAddr(t, 103)
+
+	const noUnlockDuration = int64(0)
+	const numApprovals = int64(2)
+	const txnID = int64(0)
+	var sendValue = abi.NewTokenAmount(10)
+	var nilParams = abi.MethodParams{}
+	var signers = []addr.Address{anne, bob}
+
+	builder := mock.NewBuilder(context.Background(), t, receiver).WithCaller(builtin.InitActorAddr, builtin.InitActorCodeID)
+
+	t.Run("simple propose and approval", func(t *testing.T) {
+		rt := builder.Build()
+
+		actor.verifyConstruct(rt, numApprovals, noUnlockDuration, signers...)
+
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.verifyPropose(rt, chuck, sendValue, builtin.MethodSend, nilParams)
+
+		actor.assertTransactions(rt, multisig.MultiSigTransaction{
+			To:       chuck,
+			Value:    sendValue,
+			Method:   builtin.MethodSend,
+			Params:   nilParams,
+			Approved: []addr.Address{anne},
+		})
+
+		rt.SetBalance(sendValue)
+		rt.SetCaller(bob, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		rt.ExpectSend(chuck, builtin.MethodSend, nilParams, sendValue, nil, 0)
+		actor.verifyApprove(rt, txnID)
```

Comment:
In a follow-up, this `approveAndVerify` should also verify the return value from `Approve` contains the exit code prescribed in `ExpectSend`. We should exercise both un/successful sends.